src/Symfony/Component/Messenger/Bridge/Amqp/Transport/Connection.php

```diff
@@ -516,8 +517,8 @@ public function channel(): \AMQPChannel
                     static function (): bool {
                         return false;
                     },
-                    static function (): bool {
-                        return false;
+                    static function () {
+                        throw new TransportException('Confirm calls back with nack.');
```

Comment:
The change makes sens to me but the message looks cryptic. Can we try to make it understandable to non-experts? :)

Reply:
hey @nicolas-grekas, thank you for the feedback! :)

When using this feature, it should be clear, that the channel is in confirm mode. I assume knowledge of this functionality for the formulation of the error message.

What do you think about these messages? Choose one and I will change the current one accordingly:
1. "Message publication failed due to a negative acknowledgment (nack) from the broker."
2. "An internal error occurred in the message broker, leading to unsuccessful message publication."
3. "The message publication encountered an issue as the broker returned a negative acknowledgment (nack). Therefore the message was not published."

Reply:
I like the first one

Reply:
[done](https://github.com/symfony/symfony/pull/53525/commits/03b1a1106633aa94b257b104d7e226399f329ed6) :)